src/eventmaster/plugins.go

```diff
@@ -0,0 +1,84 @@
+package main
+
+import (
+	"crypto/hmac"
+	"crypto/sha1"
+	"crypto/subtle"
+	"encoding/hex"
+	"encoding/json"
+	"net/http"
+	"time"
+)
+
+var Plugins = map[string]Plugin{}
+
+type Plugin interface {
+	ParseRequest(*http.Request, []byte) ([]*UnaddedEvent, error)
+	ValidateAuth(*http.Request, []byte) bool
+}
+
+type GitHubPlugin struct {
```

Comment:
I don't think there's a way to test the connection before an event - the only way to test is to trigger the webhook or manually send a test payload through the GitHub UI